eslint.config.mjs

```diff
@@ -0,0 +1,69 @@
+import eslint from "@eslint/js";
+import tseslint from "typescript-eslint";
+import prettierConfig from "eslint-config-prettier";
+import prettierPlugin from "eslint-plugin-prettier";
+
+export default [
+    {
+        ignores: [
+            "dist/**",
+            "**/*.d.ts",
+            "**/*.generated.ts",
+            "**/*.js",
+            "!eslint.config.js",
+        ],
+    },
+    eslint.configs.recommended,
+    ...tseslint.configs.recommended,
+    {
+        files: ["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+        languageOptions: {
+            ecmaVersion: 12,
+            sourceType: "module",
+            globals: {
+                // Node.js globals
+                console: "readonly",
+                process: "readonly",
+                Buffer: "readonly",
+                __dirname: "readonly",
+                __filename: "readonly",
+                module: "readonly",
+                require: "readonly",
+                exports: "writable",
+                global: "readonly",
+                // Jest globals
+                describe: "readonly",
+                it: "readonly",
+                test: "readonly",
+                expect: "readonly",
+                beforeEach: "readonly",
+                afterEach: "readonly",
+                beforeAll: "readonly",
+                afterAll: "readonly",
+                jest: "readonly",
+            },
```

Comment:
This can be done more elegantly using the npm "globals" package as described here: https://eslint.org/docs/latest/use/configure/language-options

Here is the globals package: https://www.npmjs.com/package/globals

There is a `globals.node` and `globals.jest` (plus others) that will more accurately include the needed global objects.

Also, the jest globals should only be included for test files.